config/config_params.go

```diff
@@ -338,6 +334,22 @@ func (config *Config) DatastoreConfig() api.CalicoAPIConfig {
 			},
 		}
 	}
+
+	// Build CalicoAPIConfig from the environment.  This means that any XxxYyy field in
+	// CalicoAPIConfigSpec can be set by a corresponding XXX_YYY or CALICO_XXX_YYY environment
+	// variable, and that the datastore type can be set by a DATASTORE_TYPE or
+	// CALICO_DATASTORE_TYPE variable.  (Except in the etcdv2 case which is handled specially
+	// above.)
+	cfg, err := client.LoadClientConfigFromEnvironment()
+	if err != nil {
+		log.WithError(err).Panic("Failed to create datastore config")
```

Comment:
Wonder if we should return an error and then shut down gracefully, presumably we'd hit this case because the config is invalid.

The panic was reasonable before because we were trying to create an empty config with no external input.  If that failed, it'd be a programming error.